### ISSUE TITLE
Cannot get syntax highlighting to work properly

### ISSUE THREAD

**Issue by briochemc** [JuliaEditorSupport/julia-vim#145]
I am having trouble figuring out why my syntax highlighting is not completely working... It seems to work fine for some things, like, e.g., for the `function` word:
![screen shot 2018-05-24 at 10 58 59 am](https://user-images.githubusercontent.com/4486578/40503069-bdec31da-5f41-11e8-9943-9ebc7ef227f5.png)
or the `while` word:
![screen shot 2018-05-24 at 10 59 29 am](https://user-images.githubusercontent.com/4486578/40503080-cac5c10a-5f41-11e8-8c39-b254b044918f.png)

But it fails on some things like in this line:
![screen shot 2018-05-24 at 10 59 35 am](https://user-images.githubusercontent.com/4486578/40503108-e5a6d1f8-5f41-11e8-8297-d07ddb03ad55.png)
where the `2` in `for p_ip = 1:2` is not colored properly, or when I define functions inline, like so:
![screen shot 2018-05-24 at 10 57 50 am](https://user-images.githubusercontent.com/4486578/40503160-036a884c-5f42-11e8-8172-fbe6536a3665.png)
where none of the functions are highlighted correctly it seems...

Am I doing something wrong?

---
FYI in case it is useful, the command `:echo b:current_syntax` returns `julia`, and the command `:syntax list` returns a lot of julia syntax-highlighting things from this repo, e.g., the first few lines are:
```
--- Syntax items ---
juliaParBlock  xxx matchgroup=juliaParDelim start=/(/ end=/)/  contains=@juliaExpressions,juliaComprehensionFor
juliaSqBraBlock xxx matchgroup=juliaParDelim start=/\[/ end=/\]/  contains=@juliaExpressions,juliaParBlockInRange,juliaRangeEnd,juli
aComprehensionFor,juliaSymbolS,juliaQuotedParBlockS,juliaQuotedQMarkParS
juliaCurBraBlock xxx matchgroup=juliaParDelim start=/{/ end=/}/  contains=@juliaExpressions  
```

**Comment by carlobaldassi** [comment]
I see the problem with the `for ip = 1:2` line (which I cannot reproduce), but I don't see what's the problem with the last example, it seems to me that everything that is supposed to be highlighted is (only operators in this case).

About the highlighting issue with `for ip = 1:2`, is it systematc? I.e. does it happen regardless of context, even in an empty file? Does it happen for all ranges or only with `for`? It might be caused by some settings in your configuration that I failed to take into account - if so, it should be easy enough to figure out. Could you verify if it works from a simplified vimrc with just julia-vim?

**Comment by briochemc** [comment]
In the last example, I thought julia-vim should highlight the function name at the start of the line (i.e., `f`) and the function name inside of the definition (i.e., `u`), like how Juno/Atom does:
![screen shot 2018-05-25 at 11 07 58 am](https://user-images.githubusercontent.com/4486578/40559573-154febe0-600c-11e8-8dc0-446e7bb4a462.png)
Is it not supposed to do so?

The `1:2` is systematic, in that it appears to only recognize the first digit as a digit. I am not sure how to proceed to do your test without erasing things I do not want to lose... Could you help me figure this out? (Sorry if that is too much to ask)

**Comment by briochemc** [comment]
I found the courage to change my `.vimrc` to find the culprit for the `1:2` issue and found that the line
```
set iskeyword+=:
```
caused the issue... (I commented it out as I do not remember why I put it there nor can I figure out what it does 😄 Hopefully this does not bite me back!) Note: I figured I should be fine to just modify the `.vimrc` as I have a `.dotfiles` repo to safeguard me from disaster:  **So thanks for making me do this - I have learned something today!** 🙂 

Otherwise, did you have any time to think about the other syntax highlighting issue I mentioned? Is it even an issue at all or is it just me? (I'm talking about the functions `f` and `u` in my example above.) By the way I also noticed that those were highlighted when checking the code in github itself, in case this helps at all.
